lib/gtfs_realtime_viz.ex

```diff
@@ -36,16 +36,35 @@ defmodule GTFSRealtimeViz do
   Renders the received protobuf files and comments into an HTML fragment that can either
   be opened directly in a browser or embedded within the HTML layout of another app.
   """
-  @spec visualize(term) :: String.t
-  def visualize(group) do
-    [vehicle_archive: vehicles_by_stop_id(group), routes: routes()]
+  @spec visualize(term, %{String.t => [[String.t]]}) :: String.t
+  def visualize(group, opts) do
+    routes = routes_we_care_about(opts)
+    vehicles_we_care_about = group
+                             |> State.vehicles
+                             |> vehicles_we_care_about(routes)
+
+    [vehicle_archive: vehicles_by_stop_id(vehicles_we_care_about), routes: opts]
     |> gen_html
     |> Phoenix.HTML.safe_to_string
   end
 
-  @spec vehicles_by_stop_id(term) :: [{String.t, %{required(String.t) => [Proto.vehicle_position]}}]
-  defp vehicles_by_stop_id(group) do
-    Enum.map(State.vehicles(group), fn {comment, vehicles} ->
+  def routes_we_care_about(opts) do
+    opts
+    |> Enum.map(fn {route, _} -> route end)
+  end
+
+  def vehicles_we_care_about(state, routes_we_care_about) do
+    Enum.map(state,
+      fn {descriptor, position_list} ->
+        filtered_positions = position_list
+        |> Enum.filter(fn position -> position.trip.route_id in routes_we_care_about end)
```

Comment:
According to the .proto definition `.trip` is optional so we can't assume that it's present. You'll have to do something like `position.trip && (position.trip.route_id in ...)`.